Vector.h

```diff
@@ -22,6 +22,10 @@ typedef struct Vector_ {
    int arraySize;
    int growthRate;
    int items;
+   /* lowest index of a soft remove/deletes, used to speed up compaction */
+   int dirty_index;
+   /* count of soft deletes, required for Vector_count to work in debug mode */
+   int dirty_count;
```

Comment:
Thoughts on "dirty_*" vs "delete_*"? I'm terrible at naming things.

I was hoping to avoid a dirty/delete count because it bloats the struct size, but it's required for `Vector_count` to be accurate. Originally, I wanted to use a `char` (no size penalty if slotted next to `owner`) that would store 3 states: 0 == no soft removes; 1 == one soft remove (so that we can shortcut to memmove in compact); 2 == more than one soft remove.

Reply:
I'm fine with `dirty_` unless any of our native speakers objects … ;-)

Also for the count: I feel more comfortable if it stores an absolute number.